src/irmin-pack/unix/ext.ml

```diff
@@ -217,9 +199,21 @@ module Maker (Config : Conf.S) = struct
             Branch.v ~fresh ~readonly path
           in
           let during_batch = false in
-          { config; contents; node; commit; branch; fm; dict; during_batch }
+          let during_gc = None in
+          {
+            config;
+            contents;
+            node;
+            commit;
+            branch;
+            fm;
+            dict;
+            during_batch;
+            during_gc;
+          }
 
         let close t =
+          if t.during_gc <> None then Errs.raise_error `Close_during_gc;
```

Comment:
This is clearly something that i going to happen in Tezos. I think we should kill (send signal 9) the child process and set `during_gc <- None`